### PR TITLE
Qa/psalm

### LINKED ISSUE #24: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `psalm.xml` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require --dev vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by superrosko**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description
 - Create a psalm.xml in the project root
 - Copy and paste the contents from this psalm.xml.dist and fix projectFiles
 - Run $ composer --dev require vimeo/psalm
 - Run $ composer require --dev psalm/plugin-phpunit
 - Run $ vendor/bin/psalm --set-baseline=psalm-baseline.xml
 - Add a composer script static-analysis with the command psalm --shepherd --stats
 - Add a new line to script: in .travis.yml: - if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi
 - Remove phpstan from the project (phpstan.neon.dist, .travis.yml entry, composer.json require-dev and scripts)

Closes #24

### CODE REVIEW

**Comment by superrosko** [comment]
@boesing, if the PRs is correct, then you can put a label hacktoberfest-accepted on it?

**Comment by boesing** [comment]
@superrosko As you can see, Travis is still failing in most (if not all) your PRs.
Please have look on what the issue is.
I think, the biggest problem is, that travis still has support for PHP 5.6, e.g.

So these Psalm PRs are blocked until we have other PRs merged which implement support for PHP 8.0 and thus dropping support for PHP <7.3.

If you have time, feel free to do another PR with dropping support for PHP <7.3 by fixing #25 in here and related issues in the other repositories. I will manage to merge the PHP 8 PR so you can rebase your psalm PRs against them afterwards 👍

**Comment by boesing** [comment]
@superrosko Please focus on psalm in this PR.
If you want to add PHP 8 support, please consider opening a new PR.
After this, we can merge that other PR and you can rebase this PR against laminas-api-tools 1.7.x.

**Please do not drop any PHP version in any Psalm related PR**

Thanks.

File: psalm.xml
Comment (boesing) [inline psalm.xml]: we should create a baseline instead

File: .travis.yml
Comment (boesing) [inline .travis.yml]: please focus on psalm integration and do not drop old PHP version.

**Review by boesing — DISMISSED** [review]

File: .travis.yml
Comment (superrosko) [reply .travis.yml]: yes, i'm turn back php support <7.1 .travis.yaml

**Comment by boesing** [comment]
As the tests will fail here, we have to wait until #25 is fixed. 👍 
Thanks for working on this PR, @superrosko! So glad the psalm integration moves forward :-)

**Comment by weierophinney** [comment]
I've rebased, and re-created the baseline, along with fixes, based on the newer Psalm version.

**Pull request merged**
